src/main/java/Application/OpenCVServer/OpenCVServerOperator.java

```diff
@@ -32,47 +29,59 @@
 
     public ServerRunner serverRunner;
     public HeartBeatRunner beatRunner;
+    private boolean connected;
 
     private static OpenCVServerOperator instance = null;
 
     private OpenCVServerOperator() {
         //TODO allow fixed values to be set externally or update system to reduce reliance on fixed file names
         fileName= "read.png";
         detector = new DetectMarker();
+        connected = false;
     }
 
     public void startConnection() throws IOException {
-        serverSocket = new ServerSocket(port);
-        clientSocket = serverSocket.accept();
-        out = new PrintWriter(clientSocket.getOutputStream(), true);
-        in = new DataInputStream(clientSocket.getInputStream());
+        if(!connected) {
+            serverSocket = new ServerSocket(port);
+            clientSocket = serverSocket.accept();
+            out = new PrintWriter(clientSocket.getOutputStream(), true);
+            in = new DataInputStream(clientSocket.getInputStream());
+            connected = true;
+        }
     }
 
     public void processing() throws IOException, IllegalArgumentException {
         currentImage = ImageIO.read(ImageIO.createImageInputStream(clientSocket.getInputStream()));
-        Mat subject = ImageProcessor.getBufferedImageMat(currentImage);
+        if(currentImage != null) {
```

Comment:
@alyhasansakr @juan-castrillon 
I'm having a problem here, when server starts before the client
if the client is connected and then fails, the server didn't immediately recognize that the connection is gone so it continues the loop with the BufferedImage being null, so when client tries to reconnect

the same does not apply when client is started first however, if Client started first and I forced it to stop, there's no problem in reconnection

I'm still researching why but any suggestions would be welcomed

Reply:
@theodorelu27 Normally the server should be the one that always waits for the client, for example the function "accept" and "read" should block the server until a client is connected. The target is to make the server autonomous, regardless of the behavior of the client, the server must always work. And that includes recognizing that the client doesn't exist or doesn't respond.

Try one more time, if still can't do it, call me and we do it together.

Reply:
@alyhasansakr  I was able to reliably stop the connection when Client fails by sending an urgent message when the input stream returns null, thus throwing a Broken Pipe Exception when Client fails

so it's basically sending a byte as a heartbeat